source/extensions/geoip_providers/maxmind/geoip_provider.h

```diff
@@ -101,12 +102,33 @@ class GeoipProvider : public Envoy::Geolocation::Driver,
   MaxmindDbPtr isp_db_;
   MaxmindDbPtr anon_db_;
   MaxmindDbPtr initMaxMindDb(const absl::optional<std::string>& db_path);
+
+  bool isLookupEnabledForDbType(const std::string& db_type) const;
+
+  void lookupInMaxmindDb(const std::string& db_type, const std::string& stat_prefix,
+                         const MaxmindDbPtr& maxmind_db,
+                         const Network::Address::InstanceConstSharedPtr& remote_address,
+                         absl::flat_hash_map<std::string, std::string>& lookup_result) const;
   void lookupInCityDb(const Network::Address::InstanceConstSharedPtr& remote_address,
                       absl::flat_hash_map<std::string, std::string>& lookup_result) const;
   void lookupInAsnDb(const Network::Address::InstanceConstSharedPtr& remote_address,
                      absl::flat_hash_map<std::string, std::string>& lookup_result) const;
   void lookupInAnonDb(const Network::Address::InstanceConstSharedPtr& remote_address,
                       absl::flat_hash_map<std::string, std::string>& lookup_result) const;
+
+  void populateGeoLookupResultForCityDb(
```

Comment:
since all `populateGeoLookupResultFor*Db` methods have very similar structure, would it be possible to refactor them into single method?